src/providers/messages.js

```diff
@@ -1,25 +1,60 @@
-export const SUBSCRIBE = {
-    'yamcs.events': (subscriptionDetails) => {
-        return `{
-            "type": "events",
-            "id": "${subscriptionDetails.subscriptionId}",
-            "options": {
-                "instance": "${subscriptionDetails.instance}"
-            }
-          }`;
-    }
+import { OBJECT_TYPES, DATA_TYPES } from '../const';
+
+
+const typeMap = {
+    [OBJECT_TYPES.EVENTS_OBJECT_TYPE]: DATA_TYPES.DATA_TYPE_EVENTS,
+    [OBJECT_TYPES.TELEMETRY_OBJECT_TYPE]: DATA_TYPES.DATA_TYPE_TELEMETRY,
+    [OBJECT_TYPES.STRING_OBJECT_TYPE]: DATA_TYPES.DATA_TYPE_TELEMETRY,
+    [OBJECT_TYPES.IMAGE_OBJECT_TYPE]: DATA_TYPES.DATA_TYPE_TELEMETRY
 };
 
-export const UNSUBSCRIBE = {
-    'yamcs.events': (subscriptionDetails) => {
-        return `{
-            "type": "cancel",
-            "options": {
-                "call": "${subscriptionDetails.call}"
-            }
-          }`;
+function buildSubscribeMessages() {
+    let subscriptionMessages = {};
+
+    for (let objectType in typeMap) {
+        if (typeMap[objectType]) {
+            let dataType = typeMap[objectType];
+
+            subscriptionMessages[objectType] = (subscriptionDetails) => {
+                let message;
+
+                if (objectType === OBJECT_TYPES.EVENTS_OBJECT_TYPE) {
+                    message = `{
+                        "type": "${dataType}",
+                        "id": "${subscriptionDetails.subscriptionId}",
+                        "options": {
+                            "instance": "${subscriptionDetails.instance}"
+                        }
+                    }`;
+                } else {
+                    message = `{
+                        "type": "${dataType}",
+                        "id": "${subscriptionDetails.subscriptionId}",
+                        "options": {
+                            "instance": "${subscriptionDetails.instance}",
+                            "processor": "realtime",
+                            "id": [{
+                                "name": "${subscriptionDetails.name}"
+                            }],
+                            "sendFromCache": false
+                        }
+                    }`;
+                }
+
+                return message;
+            };
+        }
     }
-};
 
-export const DATA_TYPE_REPLY = 'reply';
-export const DATA_TYPE_EVENTS = 'events';
+    return subscriptionMessages;
+}
+export const SUBSCRIBE = buildSubscribeMessages();
```

Comment:
This isn't being used anywhere. Or where is it being used?

Reply:
It's being used here: https://github.com/akhenry/openmct-yamcs/blob/334b5f9946c074cb4cfb493ad47e79700d45de3d/src/providers/realtime-provider.js#L103